src/components/Auth/SignUpForm.js

```diff
@@ -0,0 +1,149 @@
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [username, setUsername] = useState('');
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const referer = '/profile';
+  const auth = useAuth();
+
+  const handleSignup = e => {
+    e.preventDefault();
+    const data = {
+      username: username,
+      password: password,
+      first_name: firstName,
+      last_name: lastName,
+      email: email,
+    };
+    axios
+      .post('/auth/users/', data)
+      .then(res => {
+        auth.setAuthTokens(res.data);
+        setIsLoggedIn(true);
+        setErrorMessage('');
+      })
+      .catch(err => {
+        setErrorMessage(Object.values(err.response.data).join(''));
+      });
+  };
+  if (isLoggedIn) {
+    return <Redirect to={referer} />;
+  }
+  return (
+    <Fragment>
+      {(auth && auth.authTokens) || isLoggedIn ? (
+        <Fragment>
+          <p>Welcome!</p>
+        </Fragment>
+      ) : (
+        <Fragment>
+          <Box
+            component="form"
+            display="flex"
+            flexWrap="wrap"
+            noValidate
+            autoComplete="off"
+            onSubmit={handleSignup}
+            data-testid="signupForm"
+          >
+            <Box component="h1" fontSize={18}>
+              Create an account
+            </Box>
+            <TextField
+              id="first-name"
+              label="First Name"
+              fullWidth
+              variant="outlined"
+              margin="dense"
+              onChange={e => setFirstName(e.target.value)}
+            />
+            <TextField
+              id="last-name"
+              label="Last Name"
+              fullWidth
+              variant="outlined"
+              margin="dense"
+              onChange={e => setLastName(e.target.value)}
+            />
+            <TextField
+              id="username"
+              label="Username"
+              fullWidth
+              required
+              variant="outlined"
+              margin="dense"
+              onChange={e => setUsername(e.target.value)}
+            />
+            <TextField
+              id="email"
+              label="Email"
+              fullWidth
+              required
+              variant="outlined"
+              margin="dense"
+              type="email"
+              onChange={e => setEmail(e.target.value)}
+            />
+            <TextField
+              id="password"
+              label="Password"
+              fullWidth
+              required
+              variant="outlined"
+              margin="dense"
+              type="password"
+              onChange={e => setPassword(e.target.value)}
+            />
+
+            <Box color="error.main">{errorMessage && errorMessage}</Box>
```

Comment:
I think it would be nice to conditionally render the whole box here instead of just the `errorMessage`

Reply:
Good call! Done in 7100419.